### ISSUE TITLE
Unable to use pending block

### ISSUE THREAD

**Issue by unparalleled-js** [foundry-rs/foundry#1511]
### Component

Cast, Anvil

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (8478c08 2022-05-05T00:16:27.013152+00:00)

### What command(s) is the bug in?

cast basefee --block pending

### Operating System

macOS (amd)

### Describe the bug

I first noticed the bug when connecting `web.py` to a locally running Anvil node.
I tried to get the pending block via `web3.eth.get_block("pending")`, it failed say that there was no block with ID "pending".
I then looked up to see if "pending" was in the Book anywhere. I found it in the `cast` command `basefee`.

First, I ran:

```bash
cast basefee --block latest
```

It returned

```bash
1000000000
```

Ok, seems like this command works. So I then tried:

```bash
cast basefee --block pending
```

and it errored with:

```bash
Error: 
   0: block Number(Pending) not found

Location:
   /Users/runner/work/foundry/foundry/cast/src/lib.rs:314
```

**Comment by onbjerg** [comment]
The bug is in Anvil if it doesn't work with Web3.js as well cc @mattsse

**Comment by unparalleled-js** [comment]
I tried Postman raw RPC requests and here is what I noticed:

Step 1: I start up in Anvil.
Step 2: I call `eth_getBlockByNumber` with `"pending",false` as the args.

It returns `null` as the data.

Step 3: I then stop Anvil
Step 4: I start `npx hardhat node`
Step 5: I call `eth_getBlockByNumber` with `"pending",false` as the args... It returns `null` as the data.

It returns data with the block schema and expected (pending-like) fields.
